metakb/transform/pmkb.py

```diff
@@ -0,0 +1,480 @@
+                json.dump(output, f)
+
+    def transform(self, propositions_documents_ix=None):
+        """Transform PMKB harvested json to common data model.
+
+        :param Dict propositions_documents_ix: tracking data to properly
+            index SupportEvidence and Propositions
+        :return: Updated propositions_documents_ix object
+        """
+        data = self._extract()
+        interpretations = data['interpretations']
+        variants = {v['id']: v for v in data['variants']}
+        if not propositions_documents_ix:
+            propositions_documents_ix = {
+                # Keep track of documents index value
+                'document_index': 1,
+                # {document_id: document_index}
+                'documents': dict(),
+                # Keep track of proposition index value
+                'proposition_index': 1,
+                # {tuple: proposition_index}
+                'propositions': dict()
+            }
+
+        # build lookups for artificial IDs
+        gene_index = {}
+        disease_index = {}
+        method = self._get_methods()[0]
+        for interpretation in interpretations:
+            descriptors = self._get_descriptors(interpretation, variants,
+                                                gene_index, disease_index)
+            if not all(descriptors):
+                continue
+
+            t_descriptors, d_descriptors, g_descriptors, v_descriptors = descriptors  # noqa: E501
+            proposition = self._get_tr_proposition(v_descriptors,
+                                                   t_descriptors,
+                                                   d_descriptors,
+                                                   propositions_documents_ix)
+
+            documents = self._get_documents(interpretation,
+                                            propositions_documents_ix)
+
+            statement = self._get_statement(interpretation, proposition,
+                                            v_descriptors, t_descriptors,
+                                            d_descriptors, method,
+                                            documents)
+            self.transformed['statements'].append(statement)
+
+        self._create_json()
+        return propositions_documents_ix
+
+    def _get_descriptors(self, interpretation, variants, gene_index,
+                         disease_index):
+        """Get Descriptor objects given interpretation and associated data.
+
+        :param Dict interpretation: PMKB interpretation
+        :param Dict variants: Keys are variant labels and values are all
+            harvested variant objects
+        :param Dict gene_index: lookup gene ID from symbol
+        :param Dict disease_index: lookup disease ID from label
+        :return: Tuple containing Lists of therapy, disease, gene, and
+            variant descriptors (we expect each to be len == 1)
+        """
```

Comment:
`gene_index` and `disease_index` not used 
```suggestion
    def _get_descriptors(self, interpretation, variants):
        """Get Descriptor objects given interpretation and associated data.

        :param Dict interpretation: PMKB interpretation
        :param Dict variants: Keys are variant labels and values are all
            harvested variant objects
        :return: Tuple containing Lists of therapy, disease, gene, and
            variant descriptors (we expect each to be len == 1)
        """
```